flight/flight-core/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -213,6 +222,16 @@ public static final class Builder {
 
     /** Create the server for this builder. */
     public FlightServer build() {
+      // Get TLS info in order if the server is being configured to use mTLS.
+      try {
+        prepareTlsSettings();
+      } catch (IOException e) {
+        closeMTlsCACert();
+        closeCertChain();
+        closeKey();
```

Comment:
Honestly, this should be in a try-finally wrapping the entire build() call (since other exceptions in between this and the actual TLS settings will leak the handles). Either that, or this should be moved down to right before we use the file handles (and should still be in a try-finally)